Storage/SQL/SQLiteHistory.swift

```diff
@@ -75,21 +75,35 @@ func getLocalFrecencySQL() -> String {
 }
 
 fileprivate func escapeFTSSearchString(_ search: String) -> String {
-    // Remove double-quotes and split search string on whitespace.
-    let words = search.replacingOccurrences(of: "\"", with: "").components(separatedBy: .whitespaces)
+    // Remove double-quotes, split search string on whitespace
+    // and remove any empty strings
+    let words = search.replacingOccurrences(of: "\"", with: "").components(separatedBy: .whitespaces).filter({ !$0.isEmpty })
+
+    // If there's only one word, ensure it is longer than 2
+    // characters. Otherwise, form a different type of search
+    // string to attempt to match the start of URLs.
+    guard words.count > 1 else {
+        if let word = words.first {
+            if word.count > 2 {
```

Comment:
I changed this a bit. We only want to change the way we match strings if there are 1 or 2 characters entered *altogether*. Previously, we changed the match query *per-term*.

i.e., `a b c` will still match the old way, but `a` alone will look for `^http(s)://(www.)a*`